src/libtest/lib.rs

```diff
@@ -1928,11 +1929,7 @@ fn run_test_in_process(
         None
     };
 
-    let start = if report_time {
-        Some(Instant::now())
-    } else {
-        None
-    };
+    let start = report_time.to_option(Instant::now());
```

Comment:
```suggestion
    let start = report_time.to_option_with(Instant::now);
```
This locks a mutex so we shouldn't lock it needlessly.